src/main.ts

```diff
@@ -0,0 +1,66 @@
+import * as core from '@actions/core';
+import { exec } from '@actions/exec';
+import { which } from '@actions/io';
+import { existsSync } from 'fs';
+
+let o365CLIPath: string;
+
+async function main() {
+    try {
+        o365CLIPath = await which("o365", true);
+
+        const appFilePath: string = core.getInput("APP_FILE_PATH", { required: true });
+        const scope: string = core.getInput("SCOPE", { required: false });
+        const siteCollectionUrl: string = core.getInput("SITE_COLLECTION_URL", { required: false });
+        const skipFeatureDeployment: string = core.getInput("SKIP_FEATURE_DEPLOYMENT", { required: false }) == "true" ? "--skipFeatureDeployment" : "";
+        const overwrite: string = core.getInput("OVERWRITE", { required: false }) == "true" ? "--overwrite" : "";
+
+        let appId: string;
+
+        if (existsSync(appFilePath)) {
+            core.info("ℹ️ Starting upload and deployment...");
+            if (scope == "sitecollection") {
+                if (!siteCollectionUrl.length) {
+                    core.error("🚨 Site collection URL - SITE_COLLECTION_URL - is needed when scope is set to sitecollection.");
+                    core.setFailed("SITE_COLLECTION_URL not specified");
+                } else {
+                    appId = await executeO365CLICommand(`spo app add -p ${appFilePath} --scope sitecollection --appCatalogUrl ${siteCollectionUrl} ${overwrite}`);
+                    await executeO365CLICommand(`spo app deploy --id ${appId} --scope sitecollection --appCatalogUrl ${siteCollectionUrl} ${skipFeatureDeployment}`);
+                    await executeO365CLICommand(`spo app install --id ${appId} --siteUrl ${siteCollectionUrl} --scope sitecollection `)
```

Comment:
When `skipFeatureDeployment is true`, we will have to skip the `install` command otherwise the workflow will fail. See: https://github.com/VelinGeorgiev/spfx-react-router/runs/391059185 as reference.

Reply:
Hmm, I get an error saying that `An instance of this App already exists at the specified location.` So should we remove the install command from the code?

Reply:
@VelinGeorgiev  I think we better keep the `install` separate to `deploy`. If installing the app is needed then it can be done with the via the runscript action. The `deploy` action outputs the app id, so that can be used in runscript. What do you think?

Reply:
I added additional statement so now the code looks like:

```
if(!skipFeatureDeployment.length) {
                    await executeO365CLICommand(`spo app install --id ${appId} --siteUrl ${siteCollectionUrl} --scope sitecollection `)

}

```

Reply:
Ok. Just wanted to check when skipFeatureDeployment is false and if the app is already installed in the site collection then it will throw an error saying `App is already installed`. Do we need to handle that case?

Reply:
I couldn't catch this using your app here https://github.com/VelinGeorgiev/spfx-react-router/runs/393088800. 

I deployed and installed it multiple times and it did not failed. Maybe we should take a look closely what will happen if an increased version number of the package is deployed (something I have not tested).

Reply:
@anoopt what do you think if we remove the `install` for version 1 of the GitHub action since it might introduce unexpected behavior. After all we called the action `deploy` and it will fulfill the goal by just deploying the app at this point.

What we can do for version next is it to introduce `install` and `update` with additional action input parameter named `UPSERT` or something like this.
Then if `UPSERT` is true and if `SKIP_FEATURE_DEPLOYMENT` is false we can execute addtional logic to check if the app is due to upgrade and call `upgrade` otherwise `install` it.

If you are OK with that approach, I can create new issue in GitHub for `upsert` feature.

Reply:
Yes. That sounds good to me. Let's remove `install` from this version.

Reply:
Thank you ! I will try to publish now then.